plugins/woocommerce/includes/class-wc-post-data.php

```diff
@@ -621,6 +635,383 @@ public static function set_object_terms( $object_id, $terms, $tt_ids, $taxonomy,
+
+		$product = wc_get_product( $variation_id );
+		if ( ! $product || ! $product->is_type( 'variation' ) ) {
+			return;
+		}
+
+		$data_store = WC_Data_Store::load( 'product-variation' );
+		if ( $data_store->has_callable( 'get_attribute_summary' ) ) {
+			$new_summary     = $data_store->get_attribute_summary( $product );
+			$current_excerpt = get_post_field( 'post_excerpt', $variation_id );
+			if ( $new_summary === $current_excerpt ) {
+				return;
+			}
+
+			/**
+			* Update directly via $wpdb for performance: Avoid firing save_post hooks, loading full post objects,
+			* and creating revisions. This is safe here as we're only updating post_excerpt.
+			*/
+			$wpdb->update(
+				$wpdb->posts,
+				array( 'post_excerpt' => $new_summary ),
+				array( 'ID' => $variation_id )
+			);
+			clean_post_cache( $variation_id );
+			/**
+			* Fires after the attribute summary of a product variation has been updated.
+			*
+			* @since 10.2.0
+			* @param int $variation_id The ID of the product variation.
+			*/
+			do_action( 'woocommerce_updated_product_attribute_summary', $variation_id );
+		}
+	}
+
+	/**
+	 * Gets the threshold for synchronous regeneration of variation summaries.
+	 *
+	 * @since 10.2.0
+	 * @return int
+	 */
+	public static function get_variation_summaries_sync_threshold() {
+		/**
+		 * Filters the threshold for synchronous regeneration of variation attribute summaries.
+		 * If the number of variations affected by an update is below this threshold, the summaries
+		 * are regenerated synchronously. Otherwise, the regeneration is scheduled asynchronously.
+		 *
+		 * @since 10.2.0
+		 * @param int $threshold The default threshold value (50).
+		 * @return int The filtered threshold value.
+		 */
+		return apply_filters( 'woocommerce_regenerate_variation_summaries_sync_threshold', 50 );
+	}
+
+	/**
+	 * Handles updates to a global attribute by triggering variation summary regeneration.
+	 *
+	 * @since 10.2.0
+	 * @param int    $attribute_id Attribute ID.
+	 * @param string $attribute    Attribute name.
+	 * @param string $old_slug     Old attribute slug.
+	 */
+	public static function handle_global_attribute_updated( $attribute_id, $attribute, $old_slug ) {
+		// We use this trigger for both updates and deletions of global attributes.
+		// They pass different parameters to $old_slug - deleted attributes include the "pa_" prefix, while updated attributes do not.
+		// Remove it if existing for consistency.
+		if ( strpos( $old_slug, 'pa_' ) === 0 ) {
+			$old_slug = substr( $old_slug, 3 );
+		}
+		$taxonomy = 'pa_' . $old_slug;
+
+		global $wpdb;
+		$threshold     = self::get_variation_summaries_sync_threshold();
+		$variation_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT DISTINCT p.ID
+				FROM {$wpdb->posts} p
+				INNER JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id
+				WHERE p.post_type = 'product_variation'
+				AND pm.meta_key = %s
+				LIMIT %d",
+				'attribute_' . $taxonomy,
+				$threshold + 1
+			)
+		);
```

Comment:
@mreishus Should we use `get_posts` here? The query generated by `WP_Query` is nearly the same as this one. I don't think we need `DISTINCT` since `ID` is unique.

<img width="1537" height="850" alt="Image" src="https://github.com/user-attachments/assets/ce7ce773-17db-4271-8301-ab22f5f8a6d4" />